packages/form-layout/src/vaadin-form-item.js

```diff
@@ -255,7 +250,16 @@ class FormItem extends ThemableMixin(PolymerElement) {
     const newLabelNode = this.$.labelSlot.assignedElements()[0];
     if (newLabelNode) {
       this.__labelNode = newLabelNode;
-      this.__labelNode.id = this.__labelId;
+
+      if (this.__labelNode.id) {
+        // The new label node already has an id. Let's use it.
+        this.__labelId = this.__labelNode.id;
+      } else {
+        // The new label node doesn't have an id yet. Generate a unique one.
+        const uniqueId = (FormItem._uniqueLabelId = 1 + FormItem._uniqueLabelId || 0);
```

Comment:
Could we generate a unique id only once in the constructor?

Reply:
I did consider it but maybe it's cleaner to identify individual `<label>` instances with separate IDs? Sharing the same ID would require us to remove the id from a `<label>` element when removed/replaced with a new label, but a custom ID would need to be an exception since it should not be removed.

Reply:
Ah, right, this is reasonable. 👍